zerver/views/messages.py

```diff
@@ -910,9 +910,12 @@ def update_message_backend(request, user_profile,
         if content == "":
             raise JsonableError(_("Content can't be empty"))
         content = truncate_body(content)
-
+        ums = UserMessage.objects.filter(message=message.id)
```

Comment:
Yeah, that makes sense to me.  I need to think about it some more, but given that the choice is kind of arbitrary, I'm inclined to lock it down for now (filter on historical), and then we can loosen it up later as needed.